docs/source/traitsui_user_manual/testing/discussions/event_loop_and_exceptions.rst

```diff
@@ -0,0 +1,31 @@
+GUI event processing and exception handling
+===========================================
+
+The testing API is designed such that GUI event processing is requested
+automatically wherever there may be pending GUI events. While events are being
+processed, the global exception handling is overridden temporarily in order to
+capture any unhandled exceptions. A test error is resulted if any exceptions
+are caught.
+
+For example, |UIWrapper.locate|, |UIWrapper.perform| and |UIWrapper.inspect|
+all request pending events to be processed before and/or after interacting
+with GUI states. This automatic behavior is enabled by default, but it can be
+disabled via the |UITester.auto_process_events| flag.
```

Comment:
maybe add a cross reference to the page mentioning use with ModalDialogTester to illustrate a use case for this.  
Seeing this sentence stand alone it might be confusing why I would want to have a gui test that doesn't process any gui events

Reply:
That makes sense. I will add a reference.